### ISSUE TITLE
Partial doesn't trigger soft select in override completion

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#61439]
**Version Used**: a51b65c86bb0f42a79c47798c10ad75d5c343f92

**Steps to Reproduce**:

```cs
class Base
{
    public virtual void M() {}
}

class Derived : Base
{
    public override $$
}
```

At the `$$`, type `partial` and hit space.

**Expected Behavior**:

`partial` is left alone. I would also expect `partial` to be shown in the completion window, but afaict _no_ keywords are shown in the completion window here. A fix here will also affect `required`, as that suffers from the same problem.

**Actual Behavior**:

`partial` is replaced by whatever is selected in the completion window.

**Comment by ToddGrun** [comment]
@333fred

A couple thoughts while trying to better understand this:

1) the partial keyword on a member would require the containing class to be partial, correct?
2) C# override completion is exclusive, so I believe it prevents other providers (such as the keyword recommender) from providing values in contexts where it does.
3) With these assumptions, I can still see an issue here as completion at the following location doesn't provide 'partial':

```
partial class C
{
    public virtual $$
}
```

Would fixing 3) be sufficient, or is there additional asks here which should be addressed?

**Comment by 333fred** [comment]
1. Yes, but just because introducing partial would produce an error shouldn't mean that the IDE prevents it.
2. Sure, I believe that.
3. I _think_ that would be fine.
